cargo-ext/src/compile.rs

```diff
@@ -0,0 +1,135 @@
+use anyhow::bail;
+use cargo_metadata::Message;
+use std::path::PathBuf;
+use std::process::Command;
+use std::process::Stdio;
+use std::sync::Arc;
+
+use anyhow::Context;
+use miden_diagnostics::Verbosity;
+use midenc_session::InputFile;
```

Comment:
I generally agree with you, and I'm not really a fan of the approach used by the other Miden projects, so I've compromised by using something that is more of a blend: if it's in the same namespace, combine it, e.g. `use a::b::Foo;` and `use a::b::Bar` would get combined, but `use a::b::Foo` and `use a::c::Baz` would be separated. That generally avoids most conflicts due to changes in imports, while collapsing them considerably.

If this was my own project, I'd probably switch to grouping by item, but I wanted to at least make the attempt to follow the Miden guidelines to some degree here.

If there's a config file we can add to the project root that sets that rust-analyzer config option locally for the project, feel free to add it. Definitely annoying to have to change it globally just for the Miden projects, so if there's any way to avoid that, I'm open to it.

Reply:
Got it. It seems this import granularity is `Module`. I made an issue - https://github.com/0xPolygonMiden/compiler/issues/78